app/models/embed/purl.rb

```diff
@@ -160,6 +168,46 @@ def response
         end
     end
 
+    def file_contents
```

Comment:
This class is too big. I don't know what it's responsibilities are.  Can you extract these new methods to a new class?